src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,114 @@
+import { useState, useEffect, useRef } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Loader, PeopleTable, PeopleFilters } from '../components';
+import {
+  ERRORS,
+  getPeople,
+  getPreparedPeople,
+  getPeopleFilteredAndSorted,
+} from '../utils';
+import { Person, SearchParameters } from '../types';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get(SearchParameters.Query) || '';
+  const filterBySex = searchParams.get(SearchParameters.Sex) || '';
+  const centuries = searchParams.getAll(SearchParameters.Centuries) || [];
+  const sort = searchParams.get(SearchParameters.Sort) || '';
+  const order = searchParams.get(SearchParameters.Order) || '';
+  const firstRender = useRef(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+    const fetchPeople = async () => {
+      try {
+        const currentPeople = await getPeople();
+
+        if (!currentPeople.length) {
+          setErrorMessage(ERRORS.NO_PEOPLE_ERROR);
+        }
+
+        setPeople(getPreparedPeople(currentPeople));
+      } catch (error) {
+        setErrorMessage(ERRORS.DOWNLOAD_ERROR);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchPeople();
+  }, []);
+
+  const visiblePeople = getPeopleFilteredAndSorted(
+    filterBySex, query, centuries, sort, order, people,
+  );
+
+  useEffect(() => {
+    if (firstRender.current) {
+      firstRender.current = false;
+
+      return;
+    }
```

Comment:
Why do you do it?
Also, in this case, the user doesn't see a notification "There are no people matching the current search criteria" after page reloading.
![image](https://github.com/mate-academy/react_people-table-advanced/assets/63243136/39352535-7f33-417f-b953-8ab6e780b2ee)